BlogWebAPI/Controllers/CommentsController.cs

```diff
@@ -0,0 +1,185 @@
+                                                    });
+
+            }
+        }
+
+        /// <summary>
+        /// Method to modify the data in your own comment
+        /// </summary>
+        /// <param name="id">Identificator of the post</param>
+        /// <param name="id2">Identificator of the comment</param>
+        /// <param name="commentModel"></param>
+        /// <returns></returns>
+        [Authorize]
+        [Route("BlogApp/Posts/{id}/Comments/{id2}")]
+        [HttpPut]
+        public HttpResponseMessage EditComment(int id, int id2, [FromBody]CommentsModel commentModel)
+        {
+            try
+            {
+                int loginUserId = new UsersController().GetLoginUserId();
+                using (var blogdb = new blogdbEntities())
+                {
+                    comments newComment = (from c in blogdb.comments
+                                          where c.id_post == id && c.id == id2 && c.id_commentarist == loginUserId
+                                          select c).SingleOrDefault();
+                    if (newComment != null)
+                    {
+                        newComment.comment = commentModel.Comment;
+
+                        blogdb.Entry(newComment).State = EntityState.Modified;
+                        blogdb.SaveChanges();
+
+                        return Request.CreateResponse(HttpStatusCode.OK, "Your Comment was updated successfully.");
+                    }
+                    else
+                    {
+                        return Request.CreateErrorResponse(HttpStatusCode.NotFound,
+                        "The comment was not found to update");
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                return Request.CreateErrorResponse(HttpStatusCode.BadRequest,
+                                                    new HttpError
+                                                    {
+                                                        Message = "The comment could not be updated",
+                                                        MessageDetail = string.Format(e.Message)
+                                                    });
+            }
+        }
+
+        /// <summary>
+        /// Method to remove your own comment from a post
+        /// </summary>
+        /// <param name="id">Identificator of the post</param>
+        /// <param name="id2">Identificator of the comment</param>
+        /// <returns></returns>
+        [Authorize]
+        [Route("BlogApp/Posts/{id}/Comments/{id2}")]
+        [HttpDelete]
+        public HttpResponseMessage DeleteComment(int id, int id2)
+        {
+            try
+            {
+                int loginUserId = new UsersController().GetLoginUserId();
+                using (var blogdb = new blogdbEntities())
+                {
+                    comments commentToDelete = (from c in blogdb.comments
```

Comment:
you are using entities on controller context,  this is a business logic  and should be located in a service class